cranelift/module/src/data_context.rs

```diff
@@ -90,6 +93,14 @@ impl DataContext {
         self.description.init = Init::Bytes { contents };
     }
 
+    /// Override the segment/section for data, only supported on Object backend
+    pub fn set_section(&mut self, seg: &str, sec: &str) {
+        self.description.section = Some((
+            std::string::String::from(seg),
+            std::string::String::from(sec),
```

Comment:
idiomatic would be to just say `seg.to_owned()` rather than use `String::from`. Also, maybe this method should be named `set_segment_section`?